p2p/conn/connection.go

```diff
@@ -664,6 +692,33 @@ FOR_LOOP:
 	}
 }
 
+func (c *MConnection) recvMsgRoutine() {
+	defer c._recover()
+
+	for {
+		connMsg, ok := <-c.msgRecvQueue
+
+		if !ok {
+			break
+		}
+
+		msgLen := len(connMsg.msgBytes)
+		newBufferSize := c.msgRecvSize.Add(^uint32(msgLen - 1)) // Sub(msgLen)
+		c.Logger.Debug("Processing message", "chID", connMsg.channelID, "msgBytesBufferedLength", newBufferSize, "pendingMsgs", len(c.msgRecvQueue))
+
+		// Wake up recvRoutine if necessary
+		select {
+		case c.msgRecvAck <- struct{}{}:
```

Comment:
Couldn't it be just a bool channel instead? OR are we using struct{}{} as it is memory efficient (0 bytes) when it comes to just signalling things with channel?

Reply:
Go optimizes an empty struct not to use memory, so it is technically more memory efficient and also more idiomatic.